### ISSUE TITLE
Package request: NetworkManager-l2tp

### ISSUE THREAD

**Issue by benn13** [void-linux/void-packages#22610]
Hi,
All distros have this package in their repos and it`s very useful for people who have l2tp vpn.
https://github.com/nm-l2tp/NetworkManager-l2tp

**Comment by sgn** [comment]
For anyone want to package this:

This is copied from upstream:

> YOU MIGHT NOT BE ABLE TO PROVIDE PRE-BUILT BINARIES OF NetworkManager-l2tp >= 1.8.0 DUE TO A POTENTIAL GPL/OpenSSL LICENSE CONFLICT WITH VERSIONS OF OpenSSL < 3.0.0.

**Comment by benn13** [comment]
Could it be packaged or it has issues? is it doable?

**Comment by sgn** [comment]
- 1.2 series can be packaged
- If we switch to OpenSSL 3.0.0, we can package 1.8
- 1.8 series: I don't know.

**Comment by dkosovic** [comment]
Also from the same release notes after the ALL CAPS message :

>Please check to see if your intended Linux distribution has made any statements about considering OpenSSL (or LibreSSL) to be a "System Library" and so exempt from the licensing conflict. Alternatively a good check to determine if OpenSSL (or LibreSSL) is considered a "System Library" is to see if the postgresql package on the intended Linux distribution has dependencies on both OpenSSL (or LibreSSL) and GNU readline library

The postgresql template has both `libressl-devel` and `readline-devel` dependencies :
* https://github.com/void-linux/void-packages/blob/master/srcpkgs/postgresql/template

So it would appear Void Linux considers LibreSSL a  "System Library" as it wouldn't be possible for the postgresql package to link against both the GPL'ed GNU readline library and LibreSSL if it wasn't.

NetworkManager-l2tp 1.8.x works fine with LibreSSL, so I would recommend using LibreSSL instead of OpenSSL.

**Comment by akra80** [comment]
Can you add this package to your repositories? I think that there are many people that need this package to work, like me. 
If you can't add it, i would like to know an alternative to connect by l2tp.

Thanks.

**Comment by dkosovic** [comment]
Here is a NetworkManager-l2tp 1.8.2 package template file, if someone wants to change the maintainer field and submit it, be my guest:

```bash
# Template file for 'NetworkManager-l2tp'
pkgname=NetworkManager-l2tp
version=1.8.2
revision=1
build_style=gnu-configure
configure_args="--runstatedir=/run"
hostmakedepends="pkg-config intltool glib-devel"
makedepends="gtk+3-devel libglib-devel libsecret-devel network-manager-applet-devel
libressl-devel nss-devel ppp-devel"
depends="strongswan xl2tpd ppp"
short_desc="NetworkManager VPN plugin for L2TP/IPsec"
maintainer="Orphaned <orphan@voidlinux.org>"
license="GPL-2.0-or-later"
homepage="https://github.com/nm-l2tp/NetworkManager-l2tp"
distfiles="https://github.com/nm-l2tp/NetworkManager-l2tp/releases/download/${version}/${pkgname}-${version}.tar.xz"
checksum=1d80986dc88951e298446833d314bab7af0a933e736960d6b3b025e6b7faca94

do_configure() {
	PPP_VERSION=`sed -n 's/^\#define\s*VERSION\s*"\([^\s]*\)"$/\1/p' /usr/include/pppd/patchlevel.h`

	./configure ${configure_args} --with-pppd-plugin-dir=/usr/lib/pppd/${PPP_VERSION}
}
```

If you want to build a NetworkManager-l2tp 1.8.2 binary package now, here is a summary of how to based on the building packages instructions from :
* https://github.com/void-linux/void-packages/blob/master/README.md

Ensure that `git`, the [requirements](Requirements) from that page as well at least one of the chroot based utilities required by `xbps-src` are installed.

Clone the void-packages git repository, install the bootstrap packages and create the NetworkManager-l2tp source package directory
```
git clone git://github.com/void-linux/void-packages.git
cd void-packages
./xbps-src binary-bootstrap
```

Use a text editor to create the `srcpkgs/NetworkManager-l2tp/template` file which has the contents from the template file at the start of this message.

Build the package with:
```
./xbps-src pkg NetworkManager-l2tp
```

Once built, install the package and run time dependencies with
```
sudo xbps-install --repository=hostdir/binpkgs NetworkManager-l2tp
sudo xbps-install -S strongswan xl2tpd ppp
```

For desktop environments to fully pickup that the NetworkManager-l2tp plugins were installed, logout and re-login to the desktop environment.

For NetworkManager-l2tp  debugging which increases the pppd logging, issue:
```
sudo killall -TERM nm-l2tp-service
sudo /usr/libexec/nm-l2tp-service --debug
```
Then bring up the VPN connection in the GUI desktop environment. Not all `nm-l2tp-service --debug` logging goes to the screen, the pppd logging uses syslog and where it goes depends on which syslog daemon you are using, if for example you are using socklog, the logging would go to `/var/log/socklog/messages/current`

**Comment by ericonr** [comment]
@dkosovic any reason you wouldn't like to make PR for it?

**Comment by dkosovic** [comment]
@ericonr  I'm the upstream maintainer of NetworkManager-l2tp, I was just trying to avoid being a maintainer of a NetworkManager-l2tp package for another Linux distro on top of the ones I already maintain. I don't have anything against a PR, I just wanted to avoid being the maintainer.

**Comment by ericonr** [comment]
@dkosovic that's fair. Thanks for taking the time to make the template! I don't use it, so I wouldn't be able to maintain it properly, unfortunately. If no one takes you up on the offer, I can see about making a PR with the package already orphaned.

**Comment by benn13** [comment]
Thank a lot, I don`t have the knowledge I am just a Linux hobbyist, 
Is the template something like aur thing in arch?
can I build it?

**Comment by dkosovic** [comment]
@benn13, package template files serve a similar purpose to AUR PKGBUILDs.

I included instructions in the message that has the template file on how to build and install the NetworkManager-l2tp 1.8.2 binary package.

If no one takes up the offer of maintaining the template file, I might make a PR and maybe orphan it it a few months time.

When I was testing there was an issue with NetworkManager and dhcpcd  :
```
dhcpcd[739]: ppp0: waiting for carrier
```
it also happens with NetworkManager-pptp and probably other pppd based VPNs. I tried configuring NetworkManager to use its internal dhcp client, but didn't have luck. Although the the L2TP and IPsec connections were successful,  the DHCP failure causes a VPN connection failure. The issue may be specific to the version of NetworkManager I was using, not sure.
